plugins/sla_check.py

```diff
@@ -0,0 +1,179 @@
+"""
+This provides a way for a DataDog monitor to ping airflow and query for jobs currently
+out of compliance with their statically configured SLA. DAG's can be configured for
+SLA monitoring by adding a DAG_SLA field to them. On access, DAG's are loaded into the
+current runtime and checked for SLA's meaning they can be add/changed without requiring
+a restart of the service.
+
+DataDogs should be configured similar to the following:
+
+init_config:
+
+instances:
+  - name: Airflow SLA's
+    url: http://localhost/admin/slacheckview/healthcheck
+    timeout: 10
+
+    collect_response_time: true
+    include_content: true
+"""
+import imp
+import logging
+import os
+import re
+from datetime import datetime, date, timedelta
+
+from flask import Blueprint, jsonify
+from flask import request
+from flask_admin import BaseView, expose
+
+from airflow import settings
+from airflow.models import DagBag, DagRun
+from airflow.utils.state import State
+
+
+regex = re.compile(r'((?P<hours>\d+?)hr)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?')
```

Comment:
Name the variable more specifically.  (duration_regex, for example).  I might opt for a simpler duration format (just accept a float of hours) or standard (ISO-8601 duration syntax).  Also, i can't imagine that we would ever specify an sla in minutes or seconds (and a float would capture this).